osclients.py

```diff
@@ -47,7 +43,7 @@ class OpenStackClients(object):
     call set_keystone_version
     """
 
-    def __init__(self, auth_url=None):
+    def __init__(self, auth_url=None, modules='auto'):
```

Comment:
Why do you want to know what modules should be loaded? Perhaps we can simply load the module when it is used...